app/controllers/general.php

```diff
@@ -300,6 +301,35 @@
 
                 Authorization::setRole('role:' . Auth::USER_ROLE_APP);
                 Authorization::setDefaultStatus(false);  // Cancel security segmentation for API keys.
+
+                if (time() > ($key->getAttribute('accessedAt', 0) + (24 * 60 * 60))) {
+                    $key->setAttribute('accessedAt', time());
+                    $sdkValidator = new WhiteList([
+                        'appwrite:nodejs',
+                        'appwrite:deno',
+                        'appwrite:php',
+                        'appwrite:python',
+                        'appwrite:ruby',
+                        'appwrite:go',
+                        'appwrite:java',
+                        'appwrite:dotnet',
+                        'appwrite:dart',
+                        'appwrite:kotlin',
+                        'appwrite:swift'
+                    ], true);
+    
+                    $sdk = $request->getHeader('x-sdk-version', 'UNKNOWN');
+                    $sdk = substr($sdk, 0, strrpos($sdk, ':'));
+
+                    if ($sdkValidator->isValid($sdk)) {
+                        $sdks = $key->getAttribute('sdks', []);
+                        array_push($sdks, $sdk);
+                        $key->setAttribute('sdks', $sdks);
+                    }
+
+                    $dbForConsole->updateDocument('keys', $key->getId(), $key);
+                    $dbForConsole->deleteCachedDocument('projects', $project->getId());
```

Comment:
Hopefully this line could be removed when we move all cache cleaning logic for documents that are using subqueries into the DB filters.

Reply:
Yes agreed 👍